lib/minitest/coverage_reporter.rb

```diff
@@ -0,0 +1,118 @@
+require "json"
+
+module Minitest
+  class CoverageRunner
+
+    RESOURCE_BLACK_LIST = ['tasks']
+
+    def initialize(file_path)
+      @raw_data = JSON.load File.open file_path
+      @api_end_points = {}
+      @covered_resources = []
+      @uncovered_resources = []
+      @partially_covered_resources = {}
+    end
+
+    def raw_data
+      @raw_data
+    end
+
+    def api_end_points
+      @api_end_points
+    end
+
+    def covered_resources
+      @covered_resources
+    end
+
+    def uncovered_resources
+      @uncovered_resources
+    end
+
+    def partially_covered_resources
+      @partially_covered_resources
+    end
+
+    def uncovered_endpoints
+      api_end_points.map{ |k,v| v== false ? k : nil }.compact
+    end
+
+    def covered_endpoints
+      api_end_points.map{ |k,v| v== true ? k : nil }.compact
```

Comment:
Those do the same, couldn't they be rewritten into something like:

```ruby
    def filtered_endpoints(covered: true)
      api_endpoints.select { |_k, v| v == covered }.keys
    end
```

The name of the method can be different, it's just a suggestion.